src/opam_create.ml

```diff
@@ -0,0 +1,76 @@
+open Stdune
+
+let correct_specific
+      ((opam : Dune_project.Opam.t)
+      , { Dune_project.Opam.Package.synopsis
+        ; description
+        ; depends
+        ; conflicts
+        ; name = _
+        }) =
+  let open Opam_file.Mutator in
+  set_string "synopsis" synopsis >>>
+  set_string "description" description >>>
+  list (depends @ opam.depends)
+    (set_list "depends" Dune_project.Opam.Dependency.opam_depend) >>>
+  list (conflicts @ opam.conflicts)
+    (set_list "conflicts" Dune_project.Opam.Dependency.opam_depend)
+
+let correct project package_name =
+  let open Opam_file.Mutator in
+  opt (
+    let open Option.O in
+    let* opam = Dune_project.opam project in
+    let+ pkg = Dune_project.Opam.find opam package_name in
+    (opam, pkg))
+    correct_specific >>>
+  opt (Dune_project.license project) (set_string "license") >>>
+  list (Dune_project.authors project) (set_list "authors" mkstring) >>>
+  opt (Dune_project.version project) (set_string "version") >>>
+  opt (Option.map ~f:(Format.asprintf "%a" Dune_project.Source_kind.pp)
+         (Dune_project.source project)) (set_string "dev-repo") >>>
+  set_string "opam-version" "2.0" >>>
```

Comment:
we could also add a `x-generated-by: dune-<version>` field here into the opam file so that we can track adoption of this feature.